.github/workflows/codeql.yml

```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on: 
+  pull_request:
+  schedule:
+      - cron: '0 1 * * *'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ['python', 'java', 'javascript']
```

Comment:
For Atlas metastore, the javascript code modules (dashboarv2 and dashboardv3) are not used by product UI in any form, therefore this module are maintain by the team. We will discuss this team and update if code QL scanning is required for javascript code.
cc: @DillionTan